### PR TITLE
V2.4 release

### PR DESCRIPTION
**Pull request by stefpiatek**
## Checklist for submitter & reviewer

- [x] copy constructor copies the primary key Id of the entity
- [x] copy constructor copies all other fields of the entity, starting with `super(other)`
- [x] unique constraints added if appropriate
- [x] text definition added to columns which could have more than 255 characters
- [x] foreign key names are in the form `<entityClass>Id`, and no typos
- [x] during run, glowroot is checked for query speed and indexes added if needed
- [x] validation run completed and validated

### CODE REVIEW

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/identity/HospitalVisit.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ private HospitalVisit(HospitalVisit other) {
         this.patientClass = other.patientClass;
         this.presentationTime = other.presentationTime;
         this.sourceSystem = other.sourceSystem;
-        this.setEncounter(other.getEncounter());
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/identity/HospitalVisit.java:150]: this surprises me as a change, should we be keeping in the encounter line?

**Review by skeating — APPROVED** [review]

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/consults/ConsultationRequest.java [hunk 2]
```diff
@@ -73,12 +73,13 @@
     /**
      * \brief Date and time at which this consultationRequest was last updated.
      */
+    @Column(columnDefinition = "timestamp with time zone")
     private Instant statusChangeTime;
```
Comment (AnikaC-git) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/consults/ConsultationRequest.java:77]: In other places we call it "statusChangeDatetime"? At the moment, it's a little contradictory with the comment ...

**Review by AnikaC-git — APPROVED** [review]
Looking very well! 🥇

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/identity/HospitalVisit.java (hunk 1 above)
Comment (stefpiatek) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/identity/HospitalVisit.java:150]: haha yes it is, okay clearly I just don't have any memory! Okay great, in that case all is well!

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/consults/ConsultationRequest.java (hunk 2 above)
Comment (stefpiatek) [reply inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/consults/ConsultationRequest.java:77]: yeah that's true, You could make an issue/planner item to normalise time, datetime in InformDb?

File: inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/visit_recordings/VisitObservation.java [hunk 3]
```diff
@@ -113,14 +120,16 @@ public VisitObservation() {}
      * @param hospitalVisitId        hospital visit
      * @param visitObservationTypeId visit observation type
      * @param observationDatetime    observation datetime
+     * @param sourceSystem           information on system that last changed visit observation information
```
Comment (stefpiatek) [inline inform-db/src/main/java/uk/ac/ucl/rits/inform/informdb/visit_recordings/VisitObservation.java:123]:
```suggestion
     * @param sourceSystem           the system that last changed visit observation information
```

**Pull request merged**